test/wasi/c/read_file.c

```diff
@@ -0,0 +1,13 @@
+#include <assert.h>
+#include <stdio.h>
+
+int main() {
+  FILE* file = fopen("/sandbox/input.txt", "r");
+  assert(file != NULL);
+
+  char c = fgetc(file);
+  while (c != EOF) {
+    assert(fputc(c, stdout) != EOF);
```

Comment:
As assert() macro may being removed at compile time with -DNDEBUG, expressions that have side effects should not be put in `assert()`?

Reply:
Updated. Thanks.